src/DefinitionResolver.php

```diff
@@ -667,12 +666,26 @@ public function resolveExpressionNodeToType($expr)
                 } else {
                     $classFqn = substr((string)$t->getFqsen(), 1);
                 }
-                $fqn = $classFqn . '->' . $expr->memberName->getText($expr->getFileContents());
+                $add = '->' . $expr->memberName->getText($expr->getFileContents());
                 if ($expr->parent instanceof Node\Expression\CallExpression) {
-                    $fqn .= '()';
+                    $add .= '()';
                 }
-                $def = $this->index->getDefinition($fqn);
-                if ($def !== null) {
+                $lookupDefinition = function (string $base, string $add) use (&$lookupDefinition) {
```

Comment:
I guess it depends on the amount of data that needs to be iterated over. But my original point is that here we are looking for a single match, so that is no need to collect all parents and grandparents. 

Of course I am willing to remove the generator, if you think the gains will be negligible - just let me know how to proceed with refactoring since the method in question is in another class? 
(btw, I only included the generator because in the last attempt at fluent interfaces (a few months ago) collecting the data using ```CompletionProvider::expandParentFqns()``` was a bit slow and caused high CPU usage).